src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -133,11 +136,27 @@ private HttpResponse doApiCall(String url) {
 		}
 		return httpResponse;
 	}
-	private void getThePublishedCourseList(String url, CoursesForSubaccounts cfs) {
+	private void getThePublishedCourseList(String url, CoursesForSubaccounts cfs,HttpServletResponse response) {
 		M_log.debug("getThePublishedCourseList: Called");
 		HttpResponse httpResponse = doApiCall(url);
-		HttpEntity entity = httpResponse.getEntity();
 		ObjectMapper mapper = new ObjectMapper();
+		int statusCode = httpResponse.getStatusLine().getStatusCode();
+		if(statusCode!=200) {
```

Comment:
I think with  getting the Published courses data on separate thread will  simply this logic since i don't need to send the response then. This step of error handling will we redundant then.